src/Language/PureScript/Sugar/Names.hs

```diff
@@ -350,11 +350,8 @@ renameInModule imports (Module ss coms mn decls exps) =
       -- in scope, we throw an error.
       (Just options, _) -> do
         (mnNew, mnOrig) <- checkImportConflicts mn toName options
-        modify $ \result ->
-          M.insert
-            mnNew
-            (maybe [fmap toName qname] (fmap toName qname :) (mnNew `M.lookup` result))
-            result
+        modify $ \usedImports ->
+          M.insertWith (++) mnNew [fmap toName qname] usedImports
```

Comment:
I think it is slightly more readbale this way for newcomers to the codebase like me :)
if you don't mind the redundance, I would prefer to avoid pointfree in this particular case where the variable name helps understanding what's going on